scripts/irods/test/test_iput.py

```diff
@@ -928,6 +926,39 @@ def test_server_honors_server_config_option_checksum_read_buffer_size_in_bytes__
             # self.object_name deleted in teardown
             os.unlink(local_file)
 
+    def test_server_rejects_invalid_server_config_options_for_checksum_read_buffer_size_in_bytes__issue_7947(self):
+        self.object_name = 'test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947'
+        server_config_filename = paths.server_config_path()
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+
+        try:
+            # create a small file 
+            file_size = 10
+            local_file = os.path.join(self.admin.local_session_dir, self.object_name)
+            lib.make_file(local_file, file_size)
+
+            with lib.file_backed_up(server_config_filename):
+
+                for file_size in [-1000, -1, 0]:
```

Comment:
`file_size` -> `buffer_size`?

Should we add some overflow/underflow values? e.g. 2^64, -1*2^64

Reply:
I didn't think this would be possible because we are going through a Python layer first but I just tried "2**64" and "2**64" in Python and they gave me back a number.  I didn't know that that was possible.

Reply:
I used -2\*\*63-1 and 2\*\*63 which both are just outside the range of 64 bit signed integers.